service/controller/v11/key/key.go

```diff
@@ -50,11 +50,14 @@ const (
 	baseWorkerOverheadMultiplier = 2
 	baseWorkerOverheadModulator  = 12
 	workerIOOverhead             = "512M"
+)
 
-	// kvm endpoint annotations
-	AnnotationIp      = "endpoint.kvm.giantswarm.io/ip"
-	AnnotationService = "endpoint.kvm.giantswarm.io/service"
+const (
+	AnnotationIp         = "endpoint.kvm.giantswarm.io/ip"
+	AnnotationService    = "endpoint.kvm.giantswarm.io/service"
+	AnnotationPodDrained = "endpoint.kvm.giantswarm.io/drained"
 
+	APIEndpointAnnotation          = "giantswarm.io/api-endpoint"
```

Comment:
We have to transport the API endpoint via annotations in KVM in order to configure the node config. This is because we do not watch the kvm config but the pod itself. First versions I experimented with did lookups for the custom object but this was not nice and more fragile.